### PR TITLE
fix(migration): document table may be empty

### PR DESCRIPTION
**Pull request by laurentleseigneur**
* fix integration tests to fail when tenant has no document
* when tenant has not document to update, skip sequence update

Closes [BS-15914](https://bonitasoft.atlassian.net/browse/BS-15914)

### CODE REVIEW

File: bonita-migration-distrib/src/main/groovy/org/bonitasoft/migration/versions/v6_3_9_to_6_4_0/MigrateDocumentMysql.groovy [hunk 1]
```diff
@@ -71,16 +71,24 @@ class MigrateDocumentMysql extends MigrateDocument {
 
 
         def nextValue = sql.firstRow("SELECT MAX(doc_id) +1  AS nextValue from temp_doc").nextValue
-        sql.execute("ALTER TABLE temp_arch_doc AUTO_INCREMENT = ? ", nextValue)
 
-        sql.execute("""
+        if (hasValuesToUpdate(nextValue)) {
+            sql.execute("ALTER TABLE temp_arch_doc AUTO_INCREMENT = ? ", nextValue)
+
+            sql.execute("""
```
Comment (educhastenier) [inline bonita-migration-distrib/src/main/groovy/org/bonitasoft/migration/versions/v6_3_9_to_6_4_0/MigrateDocumentMysql.groovy]: not homogeneous with other RDBMS, insert should be outside the IF

Comment (laurentleseigneur) [inline bonita-migration-distrib/src/main/groovy/org/bonitasoft/migration/versions/v6_3_9_to_6_4_0/MigrateDocumentMysql.groovy]: will change it to conform to other RDBMS

File: bonita-migration-distrib/src/main/groovy/org/bonitasoft/migration/versions/v6_3_9_to_6_4_0/MigrateDocumentMysql.groovy [hunk 2]
```diff
@@ -111,7 +121,7 @@ class MigrateDocumentMysql extends MigrateDocument {
                         a.sourceobjectid,
                         (   SELECT count(*) +1
                             FROM arch_document_mapping b
-                            WHERE a.tenantid = a.tenantid
```
Comment (laurentleseigneur) [inline bonita-migration-distrib/src/main/groovy/org/bonitasoft/migration/versions/v6_3_9_to_6_4_0/MigrateDocumentMysql.groovy]: wrong jointure that will mix tenant results

Comment (laurentleseigneur) [inline bonita-migration-distrib/src/main/groovy/org/bonitasoft/migration/versions/v6_3_9_to_6_4_0/MigrateDocumentMysql.groovy]: fixed

**Pull request merged**